modules/actions/pool/v3/sync-pools.ts

```diff
@@ -25,19 +25,31 @@ const syncVaultData = async (
 
     // Get the data for the tables about pools
     const dbUpdates = Object.keys(onchainData).map((id) =>
-        onchainPoolUpdate(onchainData[id], allTokens, chain, id, blockNumber),
+        applyOnchainDataUpdateV3(onchainData[id], allTokens, chain, id, blockNumber),
     );
 
-    const poolsWithUSD = await poolUpsertsUsd(dbUpdates, chain, allTokens);
+    // Get the prices
+    const prices = await prisma.prismaTokenCurrentPrice
+        .findMany({
+            where: {
+                chain: chain,
+            },
+        })
+        .then((prices) => Object.fromEntries(prices.map((price) => [price.tokenAddress, price.price])));
+
+    const poolsWithUSD = dbUpdates.map((upsert) => enrichPoolUpsertsUsd(
+        { poolDynamicData: upsert.poolDynamicData, poolTokenDynamicData: upsert.poolTokenDynamicData },
+        prices,
+    ));
 
     // Update pools data to the database
     for (const { poolDynamicData, poolTokenDynamicData } of poolsWithUSD) {
         try {
             await prisma.prismaPoolDynamicData.update({
                 where: {
                     poolId_chain: {
-                        poolId: poolDynamicData.poolId,
-                        chain: poolDynamicData.chain,
```

Comment:
i think it's better to use the chain from the records rather then external, we are updating known db state

Reply:
We dont have that data here. Chain is not present in the dynamicData